src/App.js

```diff
@@ -23,21 +49,126 @@ class App extends React.Component {
+        this.setState({
+          winnerOfRound: 1,
+          numOfRounds: this.state.numOfRounds - 1,
+          playerOneScore: this.state.playerOneScore + 1,
+          message: "Player 1 wins this round!",
+        });
+      } else if (this.state.currCards[0].rank < this.state.currCards[1].rank) {
+        this.setState({
+          winnerOfRound: 2,
+          numOfRounds: this.state.numOfRounds - 1,
+          playerTwoScore: this.state.playerTwoScore + 1,
+          message: "Player 2 wins this round!",
+        });
+      } else {
+        this.setState({
+          winnerOfRound: 0,
+          numOfRounds: this.state.numOfRounds - 1,
+          message: "It's a tie!",
+        });
+      }
+    } else console.log("cards are not dealt yet.");
+  };
+
+  // react method takes two arguments
+  componentDidUpdate = (prevProp, prevState) => {
+    if (prevState.currCards !== this.state.currCards) {
+      this.checkWinner();
+    }
+  };
+
+  finalGameWinner = () => {
+    if (this.state.playerOneScore > this.state.playerTwoScore) {
+      return "Player 1 wins the game!";
+    } else if (this.state.playerOneScore < this.state.playerTwoScore) {
+      return "Player 2 wins the game!";
+    } else {
+      return "It's a game tie between player 1 & 2.";
+    }
+  };
+
+  resetGame = () => {
+    this.setState({
+      // Set default value of card deck to new shuffled deck
+      cardDeck: makeShuffledDeck(),
+      // currCards holds the cards from the current round
+      currCards: [],
+      winnerOfRound: 0,
+      message: "",
+      playerOneScore: 0,
+      playerTwoScore: 0,
+      numOfRounds: 26,
+    });
+  };
+
   render() {
-    const currCardElems = this.state.currCards.map(({ name, suit }) => (
-      // Give each list element a unique key
-      <div key={`${name}${suit}`}>
-        {name} of {suit}
-      </div>
-    ));
+    // followed along zoom recording to display cards
+    const currCardElems = this.state.currCards.map(
+      ({ name, suit, image, rank }, index) => {
+        console.log(suit);
+        const imageName = `${suit.toLowerCase()}${RANK_MAP[rank]}`;
+        console.log(imageName);
+        return (
+          // Give each list element a unique key
+          <div key={`${name}${suit}`}>
+            <b>Player {index + 1}:</b> {name} of {suit}
+            <br />
+            <br />
+            <img
+              src={imageMap[imageName]}
+              alt={`${name}${suit}`}
+              width="80px"
+            />
+          </div>
+        );
+      }
+    );
+
+    //display button text
+    let buttonText = this.state.numOfRounds === 0 ? "Reset Game" : "Deal";
 
+    // set conditional buttons
     return (
       <div className="App">
         <header className="App-header">
           <h3>High Card 🚀</h3>
           {currCardElems}
           <br />
-          <button onClick={this.dealCards}>Deal</button>
+          <button
```

Comment:
instead of using the same button for 2 different kinds of logic (1 for resetting the game, the other for dealing cards), it would be easier to maintain if you rendered 2 different buttons with their own logic. you could also then put them in their own component. e.g.:

```jsx
{
  this.state.numOfRounds // if numOfRounds is truthy
    && <button onClick={this.dealCards}>Deal</button>
    : <button onClick={this.resetGame}>Reset</button>
}
```